### ISSUE TITLE
Improve inline breakpoint experience to be similar to TypeScript's  

### ISSUE THREAD

**Issue by luabud** [microsoft/debugpy#1101]
This is coming from a conversation with @tonybaloney where he raised some issues with inline breakpoints with Python. First, inline breakpoints aren't very discoverable at the moment, it would be nice if the possible inline breakpoints were highlighted just like it's done for TypeScript:

![image](https://user-images.githubusercontent.com/45497113/185515768-ff286450-041e-44a6-9bc8-13bc0d34045b.png)

Inline breakpoints that are added to Python code also don't seem to be validated when added:
![image](https://user-images.githubusercontent.com/45497113/185515849-1a0d230b-edd6-4e5d-801b-d3e5f19ae92b.png)

**Comment by fabioz** [comment]
On a 2nd thought, maybe with https://peps.python.org/pep-0657/ we could be able to differentiate the columns on Python 3.11 (so, we'd stop at a given scope based on the context name as well as the bytecode position of the parent), in which case we could differentiate among the lambas and list comprehensions in the same line.

**Comment by tonybaloney** [comment]
See #1099 as well

**Comment by Ashark** [comment]
I tried to place inline breakpoint with `Shift + F9`, but it is treated as a usual line breakpoint.
For example, I have such code:
```python
def func1():
    return 3

def func2(arg):
    arg = arg + 5
    return arg

def func():
    return func1() * func2(4) + func1()

print(func())
pass
```

I have placed the breakpoint before the `func2(4)` invocation.
<details>
<summary>This is how it looks:</summary>

![vscode python inline breakpoint](https://github.com/microsoft/debugpy/assets/22634975/f392b4d9-b345-41c5-9f47-137f9e48ee31)
</details>

When I start debugging, the debugger is actually stopped at the beginning of line, so when I step in, it actually steps in the `func1`, but not in the `func2`.

Any way to get this working in python 3.12?

**Comment by luabud** [comment]
@ashark yes unfortunately our inline breakpoint experience doesn't support that at the moment, this is part of the feature request we're tracking here.
